File: 2022.01.20/javascript/2447.js
const n = Number(require('fs').readFileSync('/dev/stdin'));
let stars = [];
for (let i = 0 ; i<n; i++) stars.push(new Array(n));
const starFnc = (stars,r,c) => {
    if (stars[r][c] != undefined) return stars[r][c]
    if (r%3 === 1 && c%3 === 1) stars[r][c] = " ";
    else if (r>=3 && c>=3) stars[r][c] = starFnc(stars,parseInt(r/3),parseInt(c/3));
    else stars[r][c] = "*"
    return stars[r][c]
}
for (let r = 0 ; r<n ; r++){
    for (let c = 0 ; c <n ; c++){
        stars[r][c] = starFnc(stars,r,c);
    }
}
for (let i = 0 ;i<n ; i++){
  console.log(stars[i].join(""));
}

File: Boj.단계별/10_재귀/python/2447.py
n = int(input())
star_list = [[0 for _ in range(n)] for _ in range(n)]

def stars(star_list,r,c):
    if star_list[r][c] != 0 : return star_list[r][c]
    if r%3 == 1 and c%3 == 1:
        star_list[r][c] = " "
        return " "
    if r>=3 and c>= 3:
        star_list[r][c] = stars(star_list,r//3,c//3)
    else:
        star_list[r][c] = "*"
    return star_list[r][c]
for i in range(n):
    for j in range(n):
        star_list[i][j] = stars(star_list,i,j)
print('\n'.join([''.join(i) for i in star_list]))